accounts/abi/bind/base.go

```diff
@@ -152,7 +152,9 @@ func (c *BoundContract) Call(opts *CallOpts, results *[]interface{}, method stri
 		}
 	} else {
 		output, err = c.caller.CallContract(ctx, msg, opts.BlockNumber)
-		if err == nil && len(output) == 0 {
+		if err != nil {
+			return err
+		} else if len(output) == 0 {
```

Comment:
The else is unnecessary and this makes the code a tad more readable.

```suggestion
		if err != nil {
			return err
		}
		if len(output) == 0 {
```